homeassistant/components/avea/light.py

```diff
@@ -0,0 +1,97 @@
+"""Support for the Elgato Avea lights."""
+import logging
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_HS_COLOR, SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR, Light)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.util.color as color_util
+
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_AVEA = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Avea platform."""
+    import avea
+
+    nearby_bulbs = avea.discover_avea_bulbs()
+    try:
+        for bulb in nearby_bulbs:
+            bulb.get_name()
+            bulb.get_brightness()
+    except Exception:
+        raise PlatformNotReady
```

Comment:
Potential problem with this is that the exception text gets discarded, which is not nice when debugging.

Try this instead

```python
except OSError as err:
    raise PlatformNotReady from err
```

Reply:
Thanks, fixed in 20a4819fbe8fc73307da32932751ed954acb2801